packages/interfaces/src/protocols.ts

```diff
@@ -125,114 +125,156 @@ export type Callback<T extends IDecodedMessage> = (
+  UNSUPPORTED_TOPIC = "Unsupported pubsub topic",
+  TOO_MANY_REQUESTS = "Rate limit exceeded",
+  INTERNAL_ERROR = "Internal server error",
+  UNAVAILABLE = "Service temporarily unavailable",
+  NO_RLN_PROOF = "RLN proof generation failed",
+  NO_PEERS = "No relay peers available"
+}
 
-  /**
-   * Failure to find a peer with suitable protocols. This may due to a connection issue.
-   * Mitigation can be: retrying after a given time period, display connectivity issue
-   * to user or listening for `peer:connected:bootstrap` or `peer:connected:peer-exchange`
-   * on the connection manager before retrying.
-   */
+// Filter specific errors
+export enum FilterError {
+  // General errors
+  GENERIC_FAIL = "Generic error",
+  DECODE_FAILED = "Failed to decode",
   NO_PEER_AVAILABLE = "No peer available",
-
-  /**
-   * Failure to find a stream to the peer. This may be because the connection with the peer is not still alive.
-   * Mitigation can be: retrying after a given time period, or mitigation for `NO_PEER_AVAILABLE` can be used.
-   */
   NO_STREAM_AVAILABLE = "No stream available",
-
-  /**
-   * The remote peer did not behave as expected. Mitigation for `NO_PEER_AVAILABLE`
-   * or `DECODE_FAILED` can be used.
-   */
   NO_RESPONSE = "No response received",
+  STREAM_ABORTED = "Stream aborted",
 
-  //
-  // SEND ERRORS SECTION
-  //
-  /**
-   * Failure to protobuf encode the message. This is not recoverable and needs
-   * further investigation.
-   */
-  ENCODE_FAILED = "Failed to encode",
+  // Filter specific errors
+  REMOTE_PEER_REJECTED = "Remote peer rejected",
+  TOPIC_NOT_CONFIGURED = "Topic not configured",
+  SUBSCRIPTION_FAILED = "Subscription failed",
+  UNSUBSCRIBE_FAILED = "Unsubscribe failed",
+  PING_FAILED = "Ping failed",
+  TOPIC_DECODER_MISMATCH = "Topic decoder mismatch",
+  INVALID_DECODER_TOPICS = "Invalid decoder topics",
+  SUBSCRIPTION_LIMIT_EXCEEDED = "Subscription limit exceeded",
+  INVALID_CONTENT_TOPIC = "Invalid content topic",
+  PUSH_MESSAGE_FAILED = "Push message failed",
+  EMPTY_MESSAGE = "Empty message received",
+  MISSING_PUBSUB_TOPIC = "Pubsub topic missing from push message"
+}
 
-  /**
-   * The message payload is empty, making the message invalid. Ensure that a non-empty
-   * payload is set on the outgoing message.
-   */
-  EMPTY_PAYLOAD = "Payload is empty",
+// Protocol-specific failure interfaces
+export interface LightPushFailure {
+  error: LightPushError;
+  peerId?: PeerId;
+  statusCode?: number;
+  statusDesc?: string;
+  /** Protocol version used when the failure occurred (v2, v3, etc.) */
+  protocolVersion?: string;
+}
 
-  /**
-   * The message size is above the maximum message size allowed on the Waku Network.
-   * Compressing the message or using an alternative strategy for large messages is recommended.
-   */
-  SIZE_TOO_BIG = "Size is too big",
+export interface FilterFailure {
+  error: FilterError;
+  peerId?: PeerId;
+  statusCode?: number;
```

Comment:
it's good that we see it can be exposed but let's not add something that is not used 
let's keep only `error` and `peerId`

```suggestion

```